src/test/java/org/mastodon/mamut/io/exporter/graphml/GraphMLUtilsTest.java

```diff
@@ -0,0 +1,105 @@
+package org.mastodon.mamut.io.exporter.graphml;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mastodon.collection.RefCollections;
+import org.mastodon.collection.RefSet;
+import org.mastodon.mamut.feature.branch.exampleGraph.ExampleGraph4;
+import org.mastodon.mamut.model.branch.BranchLink;
+import org.mastodon.mamut.model.branch.BranchSpot;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+public class GraphMLUtilsTest
+{
+
+	private ExampleGraph4 graph4;
+
+	@Before
+	public void setUp()
+	{
+		graph4 = new ExampleGraph4();
+	}
+
+	@Test
+	public void testExportAllTracks() throws IOException
+	{
+		File tempDirectory = Files.createTempDirectory( "prefix" ).toFile();
+		String prefix = "test";
+		GraphMLUtils.exportAllTracks( graph4.getModel().getBranchGraph(), tempDirectory, prefix );
+
+		File[] files = tempDirectory.listFiles();
+		assertNotNull( files );
+		Set< String > actual = new HashSet<>( Arrays.asList( Arrays.stream( files ).map( File::getName ).toArray( String[]::new ) ) );
+
+		Set< String > expected = new HashSet<>();
+		expected.add( "test-0.graphml" );
+		expected.add( "test-4.graphml" );
+		assertEquals( expected, actual );
+	}
+
+	@Test
+	public void testExportBranches() throws IOException
+	{
+		RefSet< BranchSpot > branchSpots = RefCollections.createRefSet( graph4.getModel().getBranchGraph().vertices() );
+		branchSpots.add( graph4.branchSpotA );
+		branchSpots.add( graph4.branchSpotB );
+		branchSpots.add( graph4.branchSpotC );
+		branchSpots.add( graph4.branchSpotD );
+		branchSpots.add( graph4.branchSpotE );
+		RefSet< BranchLink > branchLinks = RefCollections.createRefSet( graph4.getModel().getBranchGraph().edges() );
+		branchLinks.add( graph4.branchLink0 );
+		branchLinks.add( graph4.branchLink1 );
+		branchLinks.add( graph4.branchLink2 );
+
+		File tempFile = Files.createTempFile( "graphml-test", "" ).toFile();
+		GraphMLUtils.exportBranches( branchSpots, branchLinks, tempFile );
+
+		// read file to string - java 8 style
+		String actualContent = new String( Files.readAllBytes( tempFile.toPath() ) );
```

Comment:
There is also a method in apache commons `FileUtils.readFileToString(file, StandardCharsets.UTF_8);`

Reply:
Thanks for the hint. I was not aware of this method. I changed it.